go/ros/ros1msg/ros1msg_parser.go

```diff
@@ -77,9 +77,13 @@ func resolveDependentFields(
 			// type with its parent package.
 			// 3. The type may be "Header". This is a special case that needs to
 			// translate to std_msgs/Header.
+
+			// Start by assuming the parent package of the field is that of the overall type.
+			fieldParentPackage := parentPackage
+
 			typeIsQualified := strings.Contains(fieldType, "/")
 			if typeIsQualified {
-				parentPackage = strings.Split(fieldType, "/")[0]
```

Comment:
the bug was, previously this would result in "parentPackage" getting modified for the next field as well - which is independent of this one.